### PR TITLE
Add option not to remove tmp directory when doing `convert_olmo_to_hf_new.py`

### PR DESCRIPTION
**Pull request by dwadden**
Attempting to remove it causes an error when using NFS.

### CODE REVIEW

File: scripts/s3_unshard_to_hf.py [hunk 1]
```diff
@@ -31,91 +31,126 @@ def make_parser():
...
+            subprocess.run(download_cmd, shell=True, check=True)
+
+        # Unshard.
+        print("Unsharding.")
+        unshard_cmd = f"python scripts/unshard.py {sharded_dir} {unsharded_dir}"
+        # Add a `--type` argument if given.
+        if args.type is not None:
+            unshard_cmd += f" --type {args.type}"
+        if args.model_only:
+            unshard_cmd += " --model-only"
```
Comment (2015aroras) [inline scripts/s3_unshard_to_hf.py:105]: nit: `model_only`

File: scripts/s3_unshard_to_hf.py
Comment (2015aroras) [inline scripts/s3_unshard_to_hf.py:1]: I'm not sure this file is supposed to be in this PR? Or at least only the part with `tmp_cleanup` is supposed to be?

Comment (dwadden) [reply scripts/s3_unshard_to_hf.py:1]: Good point, I can make a separate PR for this file. I'll tag you to review if you don't mind.

File: scripts/s3_unshard_to_hf.py (hunk 1 above)
Comment (dwadden) [reply scripts/s3_unshard_to_hf.py:105]: I was matching the hyphen in [unshard.py](https://github.com/allenai/OLMo/blob/main/scripts/unshard.py#L90), is this incorrect practice?

**Comment by dwadden** [comment]
I'll split this out into 2 separate PR's and do again.

File: scripts/s3_unshard_to_hf.py (hunk 1 above)
Comment (2015aroras) [reply scripts/s3_unshard_to_hf.py:105]: My bad, I saw you change `model-only` to `model_only` a few lines earlier so I commented without thinking much about it

**Pull request closed without merging**